synapse/handlers/oidc_handler.py

```diff
@@ -707,14 +707,26 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
             self._render_error(request, "mapping_error", str(e))
             return
 
+        # Mapping providers might not have get_extra_attributes: only call this
+        # method if it exists. It might also raise NotImplementedError.
+        extra_attributes = None
+        get_extra_attributes = getattr(
+            self._user_mapping_provider, "get_extra_attributes"
+        )
```

Comment:
This will raise an `AttributeError` if `get_extra_attributes` is not found I believe, rather than intuitively returning `None`.

Reply:
Good catch. 👍